docs/aiken-integration.md

```diff
@@ -0,0 +1,516 @@
+                                            [
+                                              i_1
+                                              [
+                                                i_0
+                                                [
+                                                  i_2
+                                                  [
+                                                    (builtin unConstrData)
+                                                    (con
+                                                      data (Constr 0 [I 69, I -69])
+                                                    )
+                                                  ]
+                                                ]
+                                              ]
+                                            ]
+                                          ]
+                                        ]
+                                        (con integer -420)
+                                      ]
+                                    )
+                                  ]
+                                  (delay (con bool False))
+                                ]
+                              )
+                            ]
+                            (delay (con unit ()))
+                          ]
+                          (delay [ (error ) (force (error )) ])
+                        ]
+                      )
+                    )
+                  )
+                )
+                (force (builtin ifThenElse))
+              ]
+            )
+            (force (force (builtin sndPair)))
+          ]
+        )
+        (force (builtin headList))
+      ]
+    )
+    (force (builtin tailList))
+  ]
+)
+```
+
+In particular, the following lines are evidence to support that the record is encoded inefficiently as `Constr 0 [<integer>, <integer>]`.
+
+```haskell
+      [
+        (builtin unConstrData)
+        (con
+          data (Constr 0 [I 69, I -69])
+        )
+      ]
+```
+
+This is awkward for LambdaBuffers since when Aiken works with the `MyRecord` type,
+it is represented with the `Constr` tag meaning that LambdaBuffers' efficient encoding would be translated to Aiken's inefficient encoding.
+Ideally, one would want to change how Aiken encodes its data types internally so that Aiken can use LambdaBuffers' efficient encoding everywhere.
+Thus, we lose all benefits of LambdaBuffers' efficient encoding when working with Aiken's mechanisms to define types because LambdaBuffers is forced to take an extra step to translate to Aiken's inefficient encoding.
+As such, Aiken's opinionated way of encoding its data is at odds with LambdaBuffers.
+
+To resolve the mismatch in the encoding of data between the two, one could alternatively sidestep all of Aiken's methods for defining types and instead use Aiken's opaque types to alias `Data` and provide ones own constructors / record accesses as follows
```

Comment:
You could also define record as usual `pub type MyRecord {data: Int}` and write a function `Data -> MyRecord`, and take datum and redeemer as `Data`, so you maintain dot record access and you do decoding only once (but not longer on demand) though you'd have to benchmark that

Reply:
I'm not quite sure I follow what we're saying here.

Are we saying:

We define
```
pub type MyRecord { data : Int }
```

then we define a function

```
fn fromData(theData : Data) -> MyRecord { ... }
```

and we write validators like
```
validator {
    pub fn hello_world(redeemer: Data, _scriptContext: Data) {
        let theRecord = fromData(redeemer)

        ...

    }
}
```

If so, this is what the text suggests in the example [here](https://github.com/mlabs-haskell/lambda-buffers/blob/jared/milestone-3-report/docs/aiken-integration.md?plain=1#L72-L146). I suppose this wasn't clear -- were we suggesting to include a sentence saying: one can encode and decode `Data` for validators in Aiken explicitly with LB's translation once at the beginning of a validator?

> you maintain dot record access and you do decoding only once

I think this is what we want avoid -- I don't see strong reasoning for using LB's efficient encoding if Aiken doesn't use it.